gitpod-io/gitpod
# [code] upgrade to VS Code 1.57.0

#### What it does

- fix https://github.com/gitpod-io/gitpod/issues/4403
- fix https://github.com/gitpod-io/gitpod/issues/4268
- fix https://github.com/gitpod-io/gitpod/issues/4130

Changes in Gitpod Code: https://github.com/gitpod-io/vscode/commit/bcd47c10d173a0c5cb174ab7451bb5133c4e9bec

#### How to test

- Start a workspace: https://ak-vscode-update.staging.gitpod-dev.com/#https://github.com/gitpod-io/gitpod
- - websockets and workers are properly proxied
    - diff editor should be operatable
    - trigger reconnection with `window.WebSocket.disconnectWorkspace()`, check that old websockets are closed and new opened of the same amount
- that user data are sync across workspaces as well as on workspace restart, especially for extensions
    - test that extensions from .gitpod.yml are installed without sync
- terminals are preserved and resized properly between window reloads
  - try to rename, change color and icon of a terminal. It should be preserved on page reload as well.
- workspace specific commands should work, i.e. F1 → type Gitpod prefix
- that a PR view is preloaded on the PR url
- test gp open and preview
- smoke test on Chrome, FF, and Safari on Mac/iPad
